File: sBuilding/Trunk/Client/sBuilding/sNonResidentComponent/Interface/INonResidentSubOrganization.cs
using sNonResidentComponent.Model;
using System.Collections.Generic;

namespace sNonResidentComponent.Interface {
    public interface INonResidentSubOrganization {
        NonResidentSubOrganization Insert(string session, NonResidentSubOrganization nonResSubOrg);
        NonResidentSubOrganization Update(string session, NonResidentSubOrganization nonResSubOrg);
        int Delete(string session, long nonSubOrgId);
        NonResidentSubOrganization Get(string session, long nonSubOrgId);
        List<NonResidentSubOrganization> GetListAllSubOrg(string session, long nonOrgId);
    }
}

File: sBuilding/Trunk/Server/Java/NonResidentService/src/com/swt/nonresident/INonResidentSubOrganization.java
package com.swt.nonresident;

import java.util.List;

import com.swt.nonresident.domain.NonResidentSubOrganization;

/**
 * Service quan ly suborg cho khach vang lai
 * 
 * @author MINH's Macbook Pro
 *
 */
public interface INonResidentSubOrganization {
	/**
	 * Them suborg
	 * 
	 * @param nonResSubOrg
	 * @return
	 */
	public NonResidentSubOrganization insert(NonResidentSubOrganization nonResSubOrg);
	
	/**
	 * Sua suborg
	 * 
	 * @param nonResSubOrg
	 * @return
	 */
	public NonResidentSubOrganization update(NonResidentSubOrganization nonResSubOrg);
	
	/**
	 * Xoa suborg
	 * 
	 * @param nonSubOrgId
	 * @return
	 */
	public int delete(long nonSubOrgId);
	
	/**
	 * Lay suborg
	 * 
	 * @param nonSubOrgId
	 * @return
	 */
	public NonResidentSubOrganization get(long nonSubOrgId);
	
	/**
	 * Lay tat ca cac danh sach suborg cua khach vang lai theo nonOrgId
	 * 
	 * @param nonOrgId
	 * @return
	 */
	public List<NonResidentSubOrganization> getListAllSubOrg(long nonOrgId);
}
